lib/src/authorization_code_grant.dart

```diff
@@ -148,7 +153,7 @@ class AuthorizationCodeGrant {
     };
 
     if (state != null) parameters['state'] = state;
-    if (!scopes.isEmpty) parameters['scope'] = scopes.join(' ');
+    if (!scopes.isEmpty) parameters['scope'] = scopes.join(delimiter ?? ' ');
```

Comment:
It would simplify the data model to assign the field to `' '` if the parameter isn't passed, rather than checking on every use.